ehcache-api/src/main/java/org/ehcache/spi/persistence/StateRepository.java

```diff
@@ -39,7 +41,7 @@ public interface StateRepository {
    * @return a state holder
    */
   @Deprecated
-  default <K extends Serializable, V extends Serializable> StateHolder<K, V> getPersistentStateHolder(String name, Class<K> keyClass, Class<V> valueClass) {
+  default <K extends Serializable, V extends Serializable> StateHolder<K, V> getPersistentStateHolder(String name, Class<K> keyClass, Class<V> valueClass) throws CachePersistenceException {
```

Comment:
Why we need a change in public api? Is this really required?

Reply:
We can avoid changing public API by removing CachePersistenceException and re-throwing as RuntimeException in FileBasedStateRepository:getPersistentStateHolder() following the ehcahe current coding practice.